network-area-diagram/src/main/java/com/powsybl/nad/svg/iidm/AbstractVoltageStyleProvider.java

```diff
@@ -108,4 +114,61 @@ protected Optional<String> getBaseVoltageStyle(BranchEdge edge, BranchEdge.Side
     }
 
     protected abstract Optional<String> getBaseVoltageStyle(Terminal terminal);
+
+    @Override
+    public List<String> getHighlightNodeStyleClasses(Node node) {
+        String subnetworkId = network.getIdentifiable(node.getEquipmentId()).getParentNetwork().getId();
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    @Override
+    public List<String> getHighlightSideEdgeStyleClasses(BranchEdge edge, BranchEdge.Side side) {
+        String subnetworkId = getSubnetworkId(edge, side);
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    @Override
+    public List<String> getHighlightThreeWtEdgStyleClasses(ThreeWtEdge edge) {
+        String subnetworkId = getSubnetworkId(edge.getEquipmentId(), edge.getSide());
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    private String getSubnetworkId(BranchEdge edge, Side side) {
+        String subnetworkId = null;
+        switch (edge.getType()) {
```

Comment:
This could be compacted, iidmSide can be reused everywhere, and terminal intermediate variable used.
```java
final TwoSides iidmSide = IidmUtils.getIidmSideFromBranchEdgeSide(side);
Terminal terminal = switch (edge.getType()) {
    case BranchEdge.LINE_EDGE -> (...)
    (...)
};
return terminal.getVoltageLevel().getParentNetwork().getId();
```